bin/diagnostic.js

```diff
@@ -1 +1,19 @@
 'use strict';
+
+const ask = require('../lib/ask.js');
+
+let answer = 7;
+
+while (answer !== '') {
+  answer = ask("Guess my secret? ");
+
+  if (answer < 1 || answer > 10) {
+    console.log('The guess needs to between 1 and 10!');
+  }
+  console.log(answer);
+
+  if (answer == 8) {
+   console.log("You guessed it!");
+   break;
+  }
+}
```

Comment:
Very close! Your approach is right on target, but some of the details here are off:
1) What happens when the user inputs a single empty space? The while loop condition evaluates to `false`, and the program stops entirely. This is what we call an "edge case" - you want to test your code with all possible cases. 
2) Multiple if-statements are useful when testing for properties that are independent of each other. That's not really the case here, but you are checking the value of `answer` in both cases. You could instead use the `else` clause. What you did not _wrong_, but it did not serve your purpose well.